### PR TITLE
Add missing type definitions

### PR DESCRIPTION
**Pull request by Maistho**
The tests and build seem to fail for me without adding type definitions for `tempy` and `mutexify`.

### CODE REVIEW

File: src/adapters/JSONFile.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import * as fs from 'fs'
-import * as mutexify from 'mutexify'
+import mutexify = require('mutexify')
```
Comment (typicode) [inline src/adapters/JSONFile.ts]: typo?

**Comment by typicode** [comment]
Thanks! :)

File: src/adapters/JSONFile.ts (hunk 1 above)
Comment (Maistho) [reply src/adapters/JSONFile.ts]: Actually no, this is the recommended way to do this. I was researching how to write type definitions in a good way when I came across this:
 
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
> When exporting a module using export =, TypeScript-specific import module = require("module") must be used to import the module.

I read more about it here: https://stackoverflow.com/a/39415662

It's possible to fake it using a namespace merged with the function declaration but this is not encouraged by DefinitelyTyped amongst others: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#should-i-add-an-empty-namespace-to-a-package-that-doesnt-export-a-module-to-use-es6-style-imports

**Comment by typicode** [comment]
After some thoughts, personally I'd prefer to go with a simple `global.d.ts` file in `src`.

It seems that this is sufficient to make the build pass:

```ts
declare module 'mutexify'
```

While the PR is good and very detailed, I prefer to avoid having to maintain typings for mutexify. Also personally, I'd not include `@types/tempy` since it doesn't match the tempy version used in the project.

I'll close this PR, but keep it as a reference if needed.

@Maistho do yo want to make the PR to add `global.d.ts`?

**Comment by Maistho** [comment]
Sure! I'll probably make a PR to DefinitelyTyped to get in the mutexify typing there as well

**Pull request closed without merging**